backend/controller/artefacts/oci_registry.go

```diff
@@ -122,45 +118,101 @@ func (s *ContainerService) GetDigestsKeys(ctx context.Context, digests []sha256.
 }
 
 // Upload uploads the specific artifact as a raw blob and links it to a manifest to prevent GC
-func (s *ContainerService) Upload(ctx context.Context, artefact Artefact) (sha256.SHA256, error) {
+func (s *OCIArtifactService) Upload(ctx context.Context, artefact Artefact) (sha256.SHA256, error) {
 	repo, err := s.repoFactory()
+	logger := log.FromContext(ctx)
 	if err != nil {
-		return sha256.SHA256{}, fmt.Errorf("unable to connect to repository '%s': %w", s.host, err)
+		return sha256.SHA256{}, fmt.Errorf("unable to connect to repository '%s': %w", s.repository, err)
 	}
-	desc := content.NewDescriptorFromBytes("application/x-octet-stream", artefact.Content)
-	if err = repo.Push(ctx, desc, bytes.NewReader(artefact.Content)); err != nil {
-		return sha256.SHA256{}, fmt.Errorf("unable to upload module blob to repository: %w", err)
+
+	// 2. Pack the files and tag the packed manifest
+	artifactType := "application/vnd.ftl.artifact"
+
+	store := memory.New()
+	desc, err := pushBlob(ctx, artifactType, artefact.Content, store)
+	if err != nil {
+		return sha256.SHA256{}, fmt.Errorf("unable to push to in memory repository %w", err)
 	}
-	return artefact.Digest, nil
-}
+	tag := desc.Digest.Hex()
+	parseSHA256, err := sha256.ParseSHA256(tag)
+	if err != nil {
+		return sha256.SHA256{}, fmt.Errorf("unable to parse sha %w", err)
+	}
+	artefact.Digest = parseSHA256
+	logger.Debugf("Tagging module blob with digest '%s'", tag)
 
-func (s *ContainerService) Download(ctx context.Context, digest sha256.SHA256) (io.ReadCloser, error) {
-	ref := createModuleRepositoryPathFromDigest(digest)
-	registry, err := s.repoFactory()
+	fileDescriptors := []ocispec.Descriptor{desc}
+	configBlob := []byte("")
```

Comment:
NBD but this would be better declared as:

```suggestion
	var configBlob []byte
```

It reduces allocations a bit.